src/SFA.DAS.ApprenticeFeedback.Application/Commands/CreateExitSurvey/CreateExitSurveyCommandHandler.cs

```diff
@@ -0,0 +1,58 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticeFeedback.Domain.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.Commands.CreateExitSurvey 
+{ 
+    public class CreateExitSurveyCommandHandler : IRequestHandler<CreateExitSurveyCommand, CreateExitSurveyCommandResponse>
+    {
+        private readonly IApprenticeFeedbackTargetContext _apprenticeFeedbackTargetContext;
+        private readonly IExitSurveyContext _exitSurveyContext;
+        private readonly IDateTimeHelper _dateTimeHelper;
+
+        public CreateExitSurveyCommandHandler(IApprenticeFeedbackTargetContext apprenticeFeedbackTargetDataContext,
+            IExitSurveyContext exitSurveyContext,
+            IDateTimeHelper dateTimeHelper)
+        {
+            _apprenticeFeedbackTargetContext = apprenticeFeedbackTargetDataContext;
+            _exitSurveyContext = exitSurveyContext;
+            _dateTimeHelper = dateTimeHelper;
+        }
+
+
+        public async Task<CreateExitSurveyCommandResponse> Handle(CreateExitSurveyCommand request, CancellationToken cancellationToken)
+        {
+            var apprenticeFeedbackTarget = await _apprenticeFeedbackTargetContext.FindById(request.ApprenticeFeedbackTargetId);
+            if (apprenticeFeedbackTarget == null)
```

Comment:
If no apprentice feedback target is found for a given command, should we return null, or should we throw an Exception ( I now understand why you returned not found, as your exposing the fact we didn't find the target so it's a not found ) That said the target should have been found to create the initial Start Survey page in the first place. So i'd still potentially lean towards throwing an InvalidOperationException instead?